package.json

```
@@ -19,7 +19,7 @@
     "build:esbuild": "pnpm run -r clean && ts-node-esm --transpileOnly .esbuild/build.ts",
     "build:mermaid": "pnpm build:esbuild --mermaid",
     "build:viz": "pnpm build:esbuild --visualize",
-    "build:types": "tsc -p ./packages/parser/tsconfig.json --emitDeclarationOnly && tsc -p ./packages/mermaid/tsconfig.json --emitDeclarationOnly && tsc -p ./packages/mermaid-zenuml/tsconfig.json --emitDeclarationOnly && tsc -p ./packages/mermaid-example-diagram/tsconfig.json --emitDeclarationOnly",
+    "build:types": "ts-node-esm --transpileOnly .build/types.ts",
```

What about:

```suggestion
    "build:types": "pnpm --recursive exec tsc --emitDeclarationOnly",
```

Very simple, and it runs in parallel (by default it's 4 jobs, see https://pnpm.io/cli/recursive#--workspace-concurrency).

The only potential issue is that 4 jobs might use up a lot of RAM (does tsc use up a lot of RAM?), but we could lower this with `--workspace-concurrency=2` or something.

---

Oooh, that makes sense then!